pandas/core/ops/array_ops.py

```diff
@@ -52,13 +51,15 @@ def comp_method_OBJECT_ARRAY(op, x, y):
         if isinstance(y, (ABCSeries, ABCIndex)):
             y = y.values
 
-        result = libops.vec_compare(x.ravel(), y, op)
+        if x.shape != y.shape:
+            raise ValueError("Shapes must match", x.shape, y.shape)
+        result = libops.vec_compare(x.ravel(), y.ravel(), op)
```

Comment:
i would add an assert / raise in vec_compare about equal shapes

Reply:
vec_compare already has this check